src/plugins/intel_gpu/src/graph/include/reorder_inst.h

```diff
@@ -86,7 +86,12 @@ class typed_primitive_inst<reorder> : public typed_primitive_inst_base<reorder>
     bool has_mean() const { return !get_typed_desc<reorder>()->mean.empty(); }
 
     void update_output_memory() override;
-    bool requires_reinterpret() const { return _req_reinterpr; }
+    bool requires_reinterpret() const {
+        if (input_memory().get_layout() != _impl_params->get_output_layout()) {
+            return true;
```

Comment:
```suggestion
            _req_reinterpr = true;
```
Is this possible to change it as above for consistency?

Reply:
For static shapes, this is already checked and set at build time.
(At that time, the check using _impl_params is same as the original result too)
For dynamic shape, it should be checked at runtime using _impl_params. 
But basically in my opinion, there is no problem even if we do reinterpretion always ..

Reply:
Ah I thought you asked about consistency but you suggested just a source level change. No problem, I will apply it.